plugins/action/network.py

```diff
@@ -118,70 +119,44 @@ def _handle_backup_option(self, result, task_vars):
             backup_path = os.path.join(cwd, "backup")
         if not filename:
-            tstamp = time.strftime(
-                "%Y-%m-%d@%H:%M:%S", time.localtime(time.time())
-            )
             filename = "%s_config.%s" % (
                 task_vars["inventory_hostname"],
                 tstamp,
             )
 
         dest = os.path.join(backup_path, filename)
-        backup_path = os.path.expanduser(
-            os.path.expandvars(
-                to_bytes(backup_path, errors="surrogate_or_strict")
-            )
-        )
-
         if not os.path.exists(backup_path):
             os.makedirs(backup_path)
 
-        new_task = self._task.copy()
-        for item in self._task.args:
-            if not item.startswith("_"):
-                new_task.args.pop(item, None)
-
-        new_task.args.update(dict(content=content, dest=dest))
-        copy_action = self._shared_loader_obj.action_loader.get(
-            "copy",
-            task=new_task,
-            connection=self._connection,
-            play_context=self._play_context,
-            loader=self._loader,
-            templar=self._templar,
-            shared_loader_obj=self._shared_loader_obj,
-        )
-        copy_result = copy_action.run(task_vars=task_vars)
-        if copy_result.get("failed"):
-            result["failed"] = copy_result["failed"]
-            result["msg"] = copy_result.get("msg")
-            return
+        changed = False
+        # Do not overwrite the destination if the contents match.
+        if not os.path.exists(dest) or checksum(dest) != checksum_s(content):
+            try:
+                with open(dest, "w") as output_file:
+                    output_file.write(content)
+            except Exception as exc:
+                result["failed"] = True
+                result["msg"] = (
+                    "Could not write to destination file %s: %s"
+                    % (dest, to_text(exc))
+                )
+                return
+            changed = True
 
         result["backup_path"] = dest
-        if copy_result.get("changed", False):
-            result["changed"] = copy_result["changed"]
-
-        if backup_options and backup_options.get("filename"):
-            result["date"] = time.strftime(
-                "%Y-%m-%d",
-                time.gmtime(os.stat(result["backup_path"]).st_ctime),
-            )
-            result["time"] = time.strftime(
-                "%H:%M:%S",
-                time.gmtime(os.stat(result["backup_path"]).st_ctime),
-            )
+        result["changed"] = changed
 
-        else:
-            result["date"] = tstamp.split("@")[0]
-            result["time"] = tstamp.split("@")[1]
-            result["shortname"] = result["backup_path"][::-1].split(".", 1)[1][
-                ::-1
-            ]
-            result["filename"] = result["backup_path"].split("/")[-1]
+        result["date"], result["time"] = tstamp.split("@")
```

Comment:
IIUC this will return the timestamp when the task is executed, however, in the original behaviour it returns the timestamp when the backup file was created `time.gmtime(os.stat(result["backup_path"]).st_ctime)

Reply:
It would do one of two things depending on whether the `filename` key was provided. If it was, it would use ctime, but this was added with the `filename` option as with a provided filename the timestamp wasn't being generated to append to the file. Given the option between the two, I went with the one that is arguably the original intent, that being the timestamp of the task and not the file.

This gets extra fuzzy when the file is unchanged as well, as the ctime on the file could be from basically any time. I'm not _entirely_ clear on what the purpose of the `date` and `time` result keys are, either. The module docs don't appear to have been updated since the addition of the `filename` option and simply say `The {date,time} extracted from the backup file name` which is why this way is being preserved here. I'm open to options moving forward, though.

Reply:
Not sure if it would be considered a breaking change, but switching to an ISO 8601 date/time might be something to talk about